unified_planning/solvers/parallel.py

```diff
@@ -56,25 +58,73 @@ def _run_parallel(self, fname, *args):
                                signaling_queue, fname, *args))
             processes.append(_p)
             _p.start()
+        processes_alive = len(processes)
+        results: List[up.solvers.PlanGenerationResult] = []
+        planners_final_status: List[str] = []
+        definitive_result_found: bool = False
         while True:
+            if processes_alive == 0: # Every planner gave a result
+                break
             (idx, res) = signaling_queue.get(block=True)
+            processes_alive -= 1
             if isinstance(res, BaseException):
                 raise res
             else:
-                break
+                assert isinstance(res, up.solvers.PlanGenerationResult)
+                # If the planner is sure about the result (optimality of the result or impossibility of the problem) exit the loop
+                if res.status == up.solvers.results.SOLVED_OPTIMALLY or res.status == up.solvers.results.UNSOLVABLE_PROVEN:
```

Comment:
I would terminate also when a SATISFICING plan is returned if the problem (args[0]) has no optimality metrics.